presto-rcfile/src/main/java/com/facebook/presto/rcfile/RcFileDecoderUtils.java

```diff
@@ -102,6 +112,39 @@ private static long readVIntInternal(Slice slice, int start, int length)
         return isNegativeVInt(slice.getByte(start)) ? ~value : value;
     }
 
+    public static long findFirstSyncPosition(RcFileDataSource dataSource, long offset, long length, long syncFirst, long syncSecond)
+            throws IOException
+    {
+        requireNonNull(dataSource, "dataSource is null");
+        checkArgument(offset >= 0, "offset is negative");
+        checkArgument(length >= 1, "length must be at least 1");
+
+        Slice sync = Slices.allocate(SIZE_OF_INT + SIZE_OF_LONG + SIZE_OF_LONG);
+        sync.setInt(0, 0xFFFF_FFFF);
+        sync.setLong(SIZE_OF_INT, syncFirst);
+        sync.setLong(SIZE_OF_INT + SIZE_OF_LONG, syncSecond);
+
+        // read 1 MB chunks at a time, but only skip ahead 1 MB - SYNC_SEQUENCE_LENGTH bytes
+        // this causes a re-read of SYNC_SEQUENCE_LENGTH bytes each time, but is much simpler code
+        byte[] buffer = new byte[Ints.checkedCast(min(1 << 20, length + SYNC_SEQUENCE_LENGTH))];
```

Comment:
Although `length + SYNC_SEQUENCE_LENGTH` works. It should be `length + SYNC_SEQUENCE_LENGTH - 1` . Doing so will make this line be coherent with the number a few lines later (change suggested in another comment).